### PR TITLE
Creates script to seed studio_people table.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Also improves the `StudioPerson#add_email` method.

### CODE REVIEW

File: bin/oneoff/populate_studio_persons [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# This script creates a studio_people entry (if one does not already exist) for
+# every teacher.
+
+require_relative '../../dashboard/config/environment'
+
+TIME_NOW = DateTime.now.freeze
+
+User.with_deleted.where(user_type: User::TYPE_TEACHER).each do |teacher|
+  # If the teacher already has an associated StudioPerson, skip them.
+  next if teacher.studio_person_id
```
Comment (Hamms) [inline bin/oneoff/populate_studio_persons]: can this not be in the `where` query?

File: dashboard/app/models/studio_person.rb [hunk 2]
```diff
@@ -8,45 +8,53 @@
...
+
+      # Merge the matching studio_person (also all associated users), if any,
+      # with this studio_person.
+      if self != matching_studio_person
+        matching_studio_person.emails_as_array.each do |existing_email|
+          add_email_to_emails(existing_email)
+        end
+        User.where(studio_person_id: matching_studio_person.id).each do |user|
+          # TODO(asher): If necessary, bypass validations as many of our users do
+          # not pass our own validations.
```
Comment (Hamms) [inline dashboard/app/models/studio_person.rb:47]: worth handling this TODO now?

File: dashboard/app/models/studio_person.rb [hunk 3]
```diff
@@ -62,7 +70,9 @@ def emails_as_array
   # @param email [String] The email to associate with this studio_person.
   def add_email_to_emails(email)
     return if email.nil? || email.blank?
-    return if emails_as_array.include? email
-    self.emails = (emails_as_array << email).join(',')
+
+    normalized_email = email.strip.downcase
```
Comment (Hamms) [inline dashboard/app/models/studio_person.rb:74]: :+1:

File: bin/oneoff/populate_studio_persons [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  # The teacher does not have an associated StudioPerson, so create one,
+  # updating the User to point to the StudioPerson.
+  studio_person = StudioPerson.create(
+    created_at: TIME_NOW,
+    updated_at: TIME_NOW
+  )
+  if teacher.email.blank?
+    teacher.update(studio_person_id: studio_person.id)
```
Comment (jeremydstone) [inline bin/oneoff/populate_studio_persons:20]: Is there a reason we only set teacher.studio_person_id if email is blank? I would've thought we would do it all the time.

Comment (ashercodeorg) [reply bin/oneoff/populate_studio_persons:20]: That, and other stuff, is accomplished in `studio_person.add_email` in the other case (specifically L34 of the other file).

File: bin/oneoff/populate_studio_persons (hunk 1 above)
Comment (ashercodeorg) [reply bin/oneoff/populate_studio_persons]: Good call, I'll follow up with a commit doing that.

File: dashboard/app/models/studio_person.rb (hunk 2 above)
Comment (ashercodeorg) [reply dashboard/app/models/studio_person.rb:47]: Given that all I can do is rescue and ignore, for now, I want to see whether and where it happens. At least for now.

File: bin/oneoff/populate_studio_persons (hunk 4 above)
Comment (jeremydstone) [reply bin/oneoff/populate_studio_persons:20]: Gotcha.

**Pull request merged**